### PR TITLE
three parts for lesson_02+flow_control

### PR DESCRIPTION
**Pull request by miroslawj**
@kuszcjan please make a review

### CODE REVIEW

**Comment by kuszcjan** [comment]
Please use
>  @kuszcjan 

instead of 

> kuszcjan

**Comment by miroslawj** [comment]
ok, my mistake - I will use @kuszcjan next time

File: students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part1.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Bishop moves
+a = int(input())
+b = int(input())
+c = int(input())
+d = int(input())
+if a >= 1 and b >= 1 and c >= 1 and d >= 1 and a <= 8 and b <= 8 and c <= 8 and d <= 8:
+    if c-a == d-b or a-c == d-b :
```
Comment (kuszcjan) [inline students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part1.py]: Could apply `abs()` here

File: students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part1.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    
+# Knight move
+a = int(input())
+b = int(input())
+c = int(input())
+d = int(input())
+if a >= 1 and b >= 1 and c >= 1 and d >= 1 and a <= 8 and b <= 8 and c <= 8 and d <= 8:
+    if ((c-a == 1 or a-c == 1) and (d-b == 2 or b-d == 2)) or ((c-a == 2 or a-c == 2) and (d-b == 1 or b-d == 1)):
+        print ("YES")
```
Comment (kuszcjan) [inline students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part1.py]: It's not very readable. Please break complex if statements into multiple lines e.g:
```
if ((c-a == 1 or a-c == 1) and (d-b == 2 or b-d == 2)) or \
   ((c-a == 2 or a-c == 2) and (d-b == 1 or b-d == 1)):
```

File: students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part2.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        print (y, sep='', end='')
+    print ('')
+
+
+# Lost card
+n = int(input())
+cards_list = []
+for x in range (n-1):
+    cards_list.append(int(input()))
+for x in range (1, n+1) :
```
Comment (kuszcjan) [inline students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part2.py]: the algorithm here could be simplified. We should avoid nesting loops if not necessary.
Please check and analyze solution below:
```
N = int(input())
all_cards = list(range(1,N+1))
for i in range(1,N):
    all_cards.remove(int(input()))
print(all_cards[0])      
```

File: students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part3.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# The second maximum
+lista = []
+provide_input_data = True
+while provide_input_data :
+    a = int(input())
+    lista.append(a)
+    if a == 0:
+        provide_input_data = False    
+lista.sort()
```
Comment (kuszcjan) [inline students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part3.py]: nice use of sort() 👍

File: students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part3.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+print (lista[-2])
+
+
+# The number of elements equal to the maximum
+lista = []
+provide_input_data = True
+while provide_input_data :
```
Comment (kuszcjan) [inline students/jakielczyk_miroslaw/lesson_02_flow_control/Lesson_02_part3.py]: Could simplify while loop by removing helper variable, for example:
```
a = int(input())
while a:
    lista.append(a)
    a = int(input())
```

**Pull request merged**